### PR TITLE
[v4.2.0-rhel] Bump Buildah to v1.27.4

### PR DESCRIPTION
**Pull request by TomSweeneyRedHat**
As the title says.  Bumping to address:
CVE-2024-1753

https://issues.redhat.com/browse/RHEL-26761

[NO NEW TESTS NEEDED]

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

**Comment by TomSweeneyRedHat** [comment]
@Luap99 and @vrothberg PTAL, especially the Makefile change.  I was having all kinds of issues as Buildah is at v1.17 and Podman was at 1.16.  After bumping the go.mod to use 1.16, I ran into errors with the vendoring until I added the compact call to the tidy per Go's recommendation.

**Comment by TomSweeneyRedHat** [comment]
@cevich could you look at the Build for fedora-35 error please?  I'm guessing the CI's go is too low and needs to be bumped to 1.17.+?

File: Makefile [hunk 1]
```diff
@@ -275,7 +275,7 @@ build-all-new-commits:
 
 .PHONY: vendor
 vendor:
-	$(GO) mod tidy
+	$(GO) mod tidy -compat=1.17
```
Comment (cevich) [inline Makefile:278]: What happens without this change?

File: go.mod [hunk 2]
```diff
@@ -1,6 +1,6 @@
 module github.com/containers/podman/v4
 
-go 1.16
+go 1.17
```
Comment (cevich) [inline go.mod:3]: And this.

**Comment by TomSweeneyRedHat** [comment]
@cevich without those changes, I couldn't get the Go vendoring to be happy.  It looked like there were changes in the version of protobuf, one of the CVE fixes, that demanded a bump in go.

**Comment by cevich** [comment]
Opened https://github.com/containers/podman/pull/22232 to remove F35 & Ubuntu from CI.  Neither are relevant for this release branch.

File: Makefile (hunk 1 above)
Comment (Luap99) [reply Makefile:278]: go changed its go.mod format in newer versions, so if you try to vendor locally with something newer it creates something different from the go 1.17 version in CI thus always causing the diff checker to trip.
Obviously we need to keep making the old go version happy so the compat flag should ensure all go version will use the same format so this change makes sense.

**Comment by Luap99** [comment]
added bloat approved but you also have a trailing white space lint error.

**Comment by TomSweeneyRedHat** [comment]
Thanks @Luap99 one stinking space after `17` in the Makefile has been eradicated.

**Comment by TomSweeneyRedHat** [comment]
Happy green test buttons here.  @mheon @luap99 or others, can get a merge on this please?

**Pull request merged**